pyicloud/base.py

```diff
@@ -98,6 +121,14 @@ def authenticate(self):
             msg = 'Invalid email/password combination.'
             raise PyiCloudFailedLoginException(msg)
 
+        # Pull X-APPLE-WEB-KB cookie from cookies
+        NewWebKBCookie = next(({key:val} for key, val in req.cookies.items() if 'X-APPLE-WEB-KB' in key), None)
```

Comment:
@coddingtonbear Sorry my code is so "hacky". I'm still learning Python and really need to read up on PEP-8. The `next` basically tells Python to set the variable to `None` if it can't find the search term. There's probably a better way to do this using a `try` but here's the concept:

https://stackoverflow.com/questions/7102050/how-can-i-get-a-python-generator-to-return-none-rather-than-stopiteration

`X-APPLE-WEB-KB` has a trailing string of seemingly random numbers and letters. Hence the use of `in`.

Hope that helps!